Kickstarter-iOS/DataSources/PledgeDataSource.swift

```diff
@@ -9,16 +9,16 @@ final class PledgeDataSource: ValueCellDataSource {
     case summary
   }
 
-  func load(reward: Reward) {
+  func load(amount: Double, currency: String) {
```

Comment:
Good point.

I've actually previously passed `Reward` object and accessed its properties...but since some of the values now come from `Project` we would also have to pass `Project` and `Reward`. As soon as this happened my thinking was: `Let's reduce the complexity for now" and when we need to pass more data either add more attributes or pass entire objects.`. I'd prefer for now going this way and if we find out that the amount of information passed in is making the function attributes too big we should think about your approach from above. For now this will stay more isolated and with less dependencies on model layer.

Hope that makes sense?

Reply:
Fair enough! Thanks for elaborating 🙏